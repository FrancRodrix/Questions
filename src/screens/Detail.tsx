import { useNavigation } from "@react-navigation/core";
import React from "react";
import {
  View,
  Text,
  TouchableOpacity,
  SafeAreaView,
  StyleSheet,
} from "react-native";

interface Props {
  navigation;
}

const Detail = (props: Props) => {
  const Item = props.route.params.data;

  return (
    <SafeAreaView style={styles.mainContainer}>
      <TouchableOpacity
        style={styles.back}
        onPress={() => {
          navigation.goBack();
        }}
      >
        <Text>Back</Text>
      </TouchableOpacity>
      <View style={styles.box}>
        <Text style={{ textAlign: "center" }}>{Item.name}</Text>
        <Text style={{ textAlign: "center" }}>{Item.name_limited}</Text>
        <Text style={{ textAlign: "center" }}>{Item.designation}</Text>
      </View>
    </SafeAreaView>
  );
};

export default Detail;

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#fe4",
  },
  back: {
    position: "absolute",
    top: 60,
    left: 30,
  },
  box: {
    alignSelf: "center",
    borderWidth: 1,
    borderRadius: 10,
    padding: 20,

    backgroundColor: "#fef",
  },
});
