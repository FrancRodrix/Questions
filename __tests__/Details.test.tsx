import "react-native";
import React from "react";
import Detail from '../src/Screens/Details';
import renderer from "react-test-renderer";

const navigation = {
    navigation: jest.fn(),
    params: {
        data:{
            item:{
                capital:'Delhi',
                name:'India',
                population:'12345',
                flags:{
                    png:'url'
                }
            
            }
           
        }
    }
}
// configure({ adapter: new Adapter() });
const tree = renderer.create(<Detail route={navigation} />);
test('renders correctly', () => {
    expect(tree).toMatchSnapshot();
});