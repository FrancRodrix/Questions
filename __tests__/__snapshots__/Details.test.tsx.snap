// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 20,
        "opacity": 1,
        "position": "absolute",
        "top": 70,
      }
    }
  >
    <Text>
      GO BACK
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#eceb",
        "borderRadius": 15,
        "borderWidth": 1,
        "padding": 20,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "url",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "height": 100,
          "width": 100,
        }
      }
    />
    <Text
      style={
        Object {
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
    >
      Delhi
    </Text>
    <Text
      style={
        Object {
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
    >
      India
    </Text>
    <Text
      style={
        Object {
          "paddingTop": 10,
          "textAlign": "center",
        }
      }
    >
      12345
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "blue",
        "borderRadius": 5,
        "marginTop": 30,
        "opacity": 1,
        "paddingHorizontal": 15,
        "paddingVertical": 7,
        "width": "50%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "textAlign": "center",
        }
      }
    >
      GET WEATHER
    </Text>
  </View>
</RCTSafeAreaView>
`;
